aws/provider_test.go

```diff
@@ -8,9 +8,11 @@ import (
 	"github.com/hashicorp/terraform/helper/schema"
 	"github.com/hashicorp/terraform/terraform"
 	"github.com/terraform-providers/terraform-provider-template/template"
+	"github.com/terraform-providers/terraform-provider-tls/tls"
 )
 
 var testAccProviders map[string]terraform.ResourceProvider
+var testAccProvidersWithTLS map[string]terraform.ResourceProvider
```

Comment:
Here we add the `TLS` provider to just the new `testAccProvidersWithTLS` map, so tests that need it can explicitly use `testAccProvidersWithTLS` instead of `testAccProviders`. This way only those tests have the additional provider and the rest remain untainted